### ISSUE TITLE
Vehicles ignore safe distances

### ISSUE THREAD

**Issue by zhaotuo0815** [michele-segata/plexe-pyapi#14]
![PLATOON](https://user-images.githubusercontent.com/59630993/192148859-9b6695b3-34a5-4540-95ae-703d3e193cc1.gif)
hello,developers,why do vehicles ignore the safety distance, and some vehicles stop suddenly, and then the front and rear vehicles are piled up together as the picture shows：
![PLATOON](https://user-images.githubusercontent.com/59630993/192148859-9b6695b3-34a5-4540-95ae-703d3e193cc1.gif)

**Comment by zhaotuo0815** [comment]
![PLATOON](https://user-images.githubusercontent.com/59630993/192148838-3db5e966-6578-44a6-96ca-ead3353faedd.gif)
<img width="753" alt="problem" src="https://user-images.githubusercontent.com/59630993/192148841-03522c98-9bac-41fe-94e1-dc06b9dc8760.png">

**Comment by michele-segata** [comment]
You are injecting platoons into an on ramp but the leader, which is driven by an ACC, has no notion of stops of right of ways, as it is in the real world. Platoons entering the highway thus basically jump into the main road without looking for incoming cars, as your car would do if you had ACC and you would not switch it off. This causes the first collision and the successive pile up.

**Comment by zhaotuo0815** [comment]
> You are injecting platoons into an on ramp but the leader, which is driven by an ACC, has no notion of stops of right of ways, as it is in the real world. Platoons entering the highway thus basically jump into the main road without looking for incoming cars, as your car would do if you had ACC and you would not switch it off. This causes the first collision and the successive pile up.

thank you so much.So how to solve this problem? Change the leader car's following model? By the way, I want the platoon vehicle to be released  after reaching a certain road section. How should it be implemented?

**Comment by michele-segata** [comment]
This is the same problem in issue #9. At the end I suggest a way to do what you want to do, but you would need to modify the car following model `MSCFModel_CC`.

**Comment by zhaotuo0815** [comment]
thank you so much.so the solution is to change the leader car's car-following model? By the way, if I want the  vehicle platoon to be released after reaching a certain road segment, how should I achieve it?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "michele-segata/plexe-pyapi"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年9月26日(星期一) 下午3:27
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [michele-segata/plexe-pyapi] Vehicles ignore safe distances (Issue #14)





 
You are injecting platoons into an on ramp but the leader, which is driven by an ACC, has no notion of stops of right of ways, as it is in the real world. Platoons entering the highway thus basically jump into the main road without looking for incoming cars, as your car would do if you had ACC and you would not switch it off. This causes the first collision and the successive pile up.

**Comment by michele-segata** [comment]
> thank you so much.so the solution is to change the leader car's car-following model?

No, changing the platoon leader car following model will not work as Plexe relies on `MSCFModel_CC` to obtain leader acceleration and speed. Please read #9. There is no quick solution. You can use the same car following model and user the `DRIVER` control algorithm, which basically uses the Krauss car following model, but you need to modify `MSCFModel_CC` (and then re-compile SUMO) so that it is still capable of providing acceleration and speed to the followers, even when a "human" is driving the vehicle.

> By the way, if I want the vehicle platoon to be released after reaching a certain road segment, how should I achieve it?

Please read the dedicated thread you opened.

**Comment by zhaotuo0815** [comment]
> > thank you so much.so the solution is to change the leader car's car-following model?
> 
> No, changing the platoon leader car following model will not work as Plexe relies on `MSCFModel_CC` to obtain leader acceleration and speed. Please read #9. There is no quick solution. You can use the same car following model and user the `DRIVER` control algorithm, which basically uses the Krauss car following model, but you need to modify `MSCFModel_CC` (and then re-compile SUMO) so that it is still capable of providing acceleration and speed to the followers, even when a "human" is driving the vehicle.
> 
> > By the way, if I want the vehicle platoon to be released after reaching a certain road segment, how should I achieve it?
> 
> Please read the dedicated thread you opened.
thank you so much. But when I use
![driver](https://user-images.githubusercontent.com/59630993/196433971-07e460d0-0ac5-413b-86b1-1b6e81071654.png)
 the DRIVER following model like( plexe.set_active_controller(vid, DRIVER)), The vehicle will automatically separate, vehicles will no longer form platooning anymore.

**Comment by michele-segata** [comment]
> > thank you so much. But when I use the DRIVER following model like( plexe.set_active_controller(vid, DRIVER)), The vehicle will automatically separate, vehicles will no longer form platooning anymore.

Yes, as I said before, you will need to modify `MSCFModel_CC` (and then re-compile SUMO) so that the leader vehicle still provides acceleration and speed to the followers, even when a "human" is driving the vehicle. By default it will no work, you will need to work on the source code.

**Comment by zhaotuo0815** [comment]
> > > thank you so much. But when I use the DRIVER following model like( plexe.set_active_controller(vid, DRIVER)), The vehicle will automatically separate, vehicles will no longer form platooning anymore.
> 
> Yes, as I said before, you will need to modify `MSCFModel_CC` (and then re-compile SUMO) so that the leader vehicle still provides acceleration and speed to the followers, even when a "human" is driving the vehicle. By default it will no work, you will need to work on the source code.

Thank you so much for your replying.  The next question is how to re-compile SUMO.

**Comment by michele-segata** [comment]
> Thank you so much for your replying. The next question is how to re-compile SUMO.

Please follow the instructions on the official [SUMO website](https://sumo.dlr.de/docs/Installing/Linux_Build.html)

**Comment by michele-segata** [comment]
Closing as inactive.
